File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/objetos/Box.java
package com.nopalsoft.sokoban.objetos;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion;
import com.nopalsoft.sokoban.Assets;

public class Box extends Tiles {
    public static final int COLOR_BEIGE = 1;
    public static final int COLOR_DARK_BEIGE = -1;
    public static final int COLOR_BLACK = 2;
    public static final int COLOR_DARK_BLACK = -2;
    public static final int COLOR_BLUE = 3;
    public static final int COLOR_DARK_BLUE = -3;
    public static final int COLOR_BROWN = 4;
    public static final int COLOR_DARK_BROWN = -4;
    public static final int COLOR_GRAY = 5;
    public static final int COLOR_DARK_GRAY = -5;
    public static final int COLOR_RED = 6;
    public static final int COLOR_DARK_RED = -6;
    public static final int COLOR_YELLOW = 7;
    public static final int COLOR_DARK_YELLOW = -7;
    public static final int COLOR_PURPLE = 8;
    public static final int COLOR_DARK_PURPLE = -8;
    public boolean isInRightEndPoint;
    int numColor;
    AtlasRegion keyFrame;

    public Box(int posicion, String color) {
        super(posicion);

        isInRightEndPoint = false;

        switch (color) {
            case "brown":
                numColor = COLOR_BROWN;
                break;
            case "gray":
                numColor = COLOR_GRAY;
                break;
            case "purple":
                numColor = COLOR_PURPLE;
                break;
            case "blue":
                numColor = COLOR_BLUE;
                break;
            case "black":
                numColor = COLOR_BLACK;
                break;
            case "beige":
                numColor = COLOR_BEIGE;
                break;
            case "yellow":
                numColor = COLOR_YELLOW;
                break;
            case "red":
                numColor = COLOR_RED;
                break;
        }

        setTextureColor(numColor);
    }

    private void setTextureColor(int numColor) {
        switch (numColor) {
            case COLOR_BEIGE:
                keyFrame = Assets.cajaBeige;
                break;
            case COLOR_DARK_BEIGE:
                keyFrame = Assets.cajaDarkBeige;
                break;
            case COLOR_BLACK:
                keyFrame = Assets.cajaBlack;
                break;
            case COLOR_DARK_BLACK:
                keyFrame = Assets.cajaDarkBlack;
                break;
            case COLOR_BLUE:
                keyFrame = Assets.cajaBlue;
                break;
            case COLOR_DARK_BLUE:
                keyFrame = Assets.cajaDarkBlue;
                break;
            case COLOR_BROWN:
                keyFrame = Assets.cajaBrown;
                break;
            case COLOR_DARK_BROWN:
                keyFrame = Assets.cajaDarkBrown;
                break;
            case COLOR_GRAY:
                keyFrame = Assets.cajaGray;
                break;
            case COLOR_DARK_GRAY:
                keyFrame = Assets.cajaDarkGray;
                break;
            case COLOR_RED:
                keyFrame = Assets.cajaRed;
                break;
            case COLOR_DARK_RED:
                keyFrame = Assets.cajaDarkRed;
                break;
            case COLOR_YELLOW:
                keyFrame = Assets.cajaYellow;
                break;
            case COLOR_DARK_YELLOW:
                keyFrame = Assets.cajaDarkYellow;
                break;
            case COLOR_PURPLE:
                keyFrame = Assets.cajaPurple;
                break;
            case COLOR_DARK_PURPLE:
                keyFrame = Assets.cajaDarkPurple;
                break;
        }
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        batch.draw(keyFrame, getX(), getY(), SIZE, SIZE);
    }

    public void setIsInEndPoint(EndPoint endPoint) {
        numColor = Math.abs(numColor);
        isInRightEndPoint = false;
        if (endPoint != null && endPoint.numColor == numColor) {
            numColor = -numColor;
            isInRightEndPoint = true;
        }
        setTextureColor(numColor);

    }
}

File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/objects/Box.kt
package com.nopalsoft.sokoban.objects

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion
import com.nopalsoft.sokoban.Assets
import kotlin.math.abs

class Box(position: Int, color: String) : Tiles(position) {
    @JvmField
    var isInRightEndPoint = false
    private var numColor = when (color) {
        "brown" -> COLOR_BROWN
        "gray" -> COLOR_GRAY
        "purple" -> COLOR_PURPLE
        "blue" -> COLOR_BLUE
        "black" -> COLOR_BLACK
        "beige" -> COLOR_BEIGE
        "yellow" -> COLOR_YELLOW
        "red" -> COLOR_RED
        else -> {
            throw Exception("The provided color is not known")
        }
    }
    private lateinit var keyFrame: AtlasRegion

    init {
        setTextureColor(numColor)
    }

    private fun setTextureColor(numColor: Int) {
        keyFrame = when (numColor) {
            COLOR_BEIGE -> Assets.beigeBox
            COLOR_DARK_BEIGE -> Assets.darkBeigeBox
            COLOR_BLACK -> Assets.blackBox
            COLOR_DARK_BLACK -> Assets.darkBlackBox
            COLOR_BLUE -> Assets.blueBox
            COLOR_DARK_BLUE -> Assets.darkBlueBox
            COLOR_BROWN -> Assets.brownBox
            COLOR_DARK_BROWN -> Assets.darkBrownBox
            COLOR_GRAY -> Assets.grayBox
            COLOR_DARK_GRAY -> Assets.darkGrayBox
            COLOR_RED -> Assets.redBox
            COLOR_DARK_RED -> Assets.darkRedBox
            COLOR_YELLOW -> Assets.yellowBox
            COLOR_DARK_YELLOW -> Assets.darkYellowBox
            COLOR_PURPLE -> Assets.purpleBox
            COLOR_DARK_PURPLE -> Assets.darkPurpleBox
            else -> {
                throw Exception("The provided color is not known")
            }
        }
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        batch.draw(keyFrame, x, y, SIZE, SIZE)
    }

    fun setIsInEndPoint(endPoint: EndPoint) {
        numColor = abs(numColor)
        isInRightEndPoint = false
        if (endPoint.numColor == numColor) {
            numColor = -numColor
            isInRightEndPoint = true
        }
        setTextureColor(numColor)
    }

    companion object {
        const val COLOR_BEIGE = 1
        const val COLOR_DARK_BEIGE = -1
        const val COLOR_BLACK = 2
        const val COLOR_DARK_BLACK = -2
        const val COLOR_BLUE = 3
        const val COLOR_DARK_BLUE = -3
        const val COLOR_BROWN = 4
        const val COLOR_DARK_BROWN = -4
        const val COLOR_GRAY = 5
        const val COLOR_DARK_GRAY = -5
        const val COLOR_RED = 6
        const val COLOR_DARK_RED = -6
        const val COLOR_YELLOW = 7
        const val COLOR_DARK_YELLOW = -7
        const val COLOR_PURPLE = 8
        const val COLOR_DARK_PURPLE = -8
    }
}